component/otelcol/exporter/prometheus/internal/convert/convert.go

```diff
@@ -458,11 +460,34 @@ func (conv *Converter) consumeHistogram(app storage.Appender, memResource *memor
 			}
 		}
 
+		// Sort the histogram by bounds ascending
+		bSize := int(math.Min(float64(dp.ExplicitBounds().Len()), float64(dp.BucketCounts().Len())))
+		buckets := make(map[float64]uint64, bSize)
+		bounds := make([]float64, bSize)
+		for i := 0; i < dp.ExplicitBounds().Len() && i < dp.BucketCounts().Len(); i++ {
+			bound := dp.ExplicitBounds().At(i)
+			buckets[bound] = dp.BucketCounts().At(i)
+			bounds = append(bounds, bound)
+		}
+
+		sort.Float64s(bounds)
+
+		// Calculate cumulative count value as Prometheus wants it
```

Comment:
Do you think there's a way to reword here so that it's clearer we don't refer to cumulative _temporality_ but so that we make it clear it's about the way that Prometheus uses cumulative buckets for histograms?

Reply:
Done! Thanks.